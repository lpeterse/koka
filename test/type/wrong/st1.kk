val r = ref([])

fun wrong() {
  r := [1]
  if ((!r).head) then 1 else 0
}