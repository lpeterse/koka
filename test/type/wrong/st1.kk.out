test/type/wrong/st1.kk(1, 9): error: effects do not match
 context : ref([])
 term : ref
 inferred effect: <alloc<_h>|_e>
 expected effect: total
 because : Generalized values cannot have an effect